examples/boston_housing_meta_optimization.py

```diff
@@ -94,16 +94,19 @@
 
 print("Meta-fitting on train:")
 p = p.meta_fit(X_train, y_train, metastep=RandomSearch(
+    p,
```

Comment:
I'm against adding this argument as it was using `.set_step` before properly. It feels weird to have to pass p to the object that it will act on here, because it's the `meta_fit` method's responsibility to set p as self to ensure p isn't something different, which would be weird.